### ISSUE TITLE
Misleading error message for 

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#55883]
Take the following code:

```ts
export abstract class YaddaBase {
    protected roots = "hi";
}

export class DerivedYadda extends YaddaBase {
    public get rootTests() {
        return super.roots;
    }
}
```

TypeScript nightly (5.3.0-dev.20230926) reports:

> Class field 'roots' defined by the parent class is not accessible in the child class via super.

The error message here is confusing - one might believe that this has to do with accessing a member with a protected modifier, but it is really about accessing an instance field with `super`. A better error message would be:

> 'roots' is defined as an instance property and must be accessed through 'this', not 'super'.

We can also provide a quick fix for this error message.

**Comment by andrewbranch** [comment]
https://github.com/microsoft/TypeScript/pull/54056

**Comment by sandersn** [comment]
How is

> Class field 'roots' is defined by the super class and must be accessed through 'this', not 'super'.

I don't understand the advantage of calling 'roots' a class field vs instance property, so maybe "Instance property 'roots' is ..." is better.
I definitely think 'super class' is more accurate than 'parent class' and more *a propos* than 'base class' since `super` is the mistaken keyword in question.

**Comment by DanielRosenwasser** [comment]
Because it's not really about class fields, it's about instance properties.

Though for what it's worth, we don't issue the error in JavaScript for properties inferred from assignments.

```js
// @ts-check

class YaddaBase {
    constructor() {
        this.roots = "hi";
    }
}

class DerivedYadda extends YaddaBase {
    get rootTests() {
        return this.roots; // no error?
    }
}

console.log(new DerivedYadda().rootTests); // prints undefined!
```

**Comment by fatcerberus** [comment]
I prefer “instance property”, as that is the important distinction: it’s not accessible through `super` precisely *because* it’s a property of the instance and not the prototype.

**Comment by fatcerberus** [comment]
Wait, why does that print undefined?  Shouldn’t the base class ctor run before the getter?

**Comment by DanielRosenwasser** [comment]
Sorry, fixed now to make it `super.roots` again.
